pkg/k8shandler/elasticsearch.go

```diff
@@ -623,6 +623,34 @@ func updateIndexReplicas(clusterName, namespace string, client client.Client, in
 	return (payload.StatusCode == 200 && acknowledged), payload.Error
 }
 
+func ensureTokenHeader(header http.Header) http.Header {
+
+	if header == nil {
+		header = map[string][]string{}
+	}
+
+	if saToken, ok := readSAToken(); ok {
+		header["x-forwarded-access-token"] = []string{
+			saToken,
+		}
+	}
+
+	return header;
+}
+
+// we want to read each time so that we can be sure to have the most up to date
+// token in the case where our perms change and a new token is mounted
+func readSAToken() (string, bool) {
+	// read from /var/run/secrets/kubernetes.io/serviceaccount/token
+	token, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	if err != nil {
+		logrus.Errorf("Unable to read file to get contents: %v", err)
```

Comment:
```go
filename := "/var/run/secrets/kubernetes.io/serviceaccount/token"
token, err := ioutil.ReadFile(filename)
if err != nil {
    logrus.Errorf("Unable to read auth token from file [%s]: %v", filename, err)
    return "", false
}
if token == nil || token == "" {
    logrus.Errorf("Unable to read auth token from file [%s]: empty token", filename)
    return "", false
}
return string(token), true
}
```

Reply:
i dont think `token` can ever be `nil` since `ReadFile` does not return a pointer to a byte array.
Updated to check for an empty token though